WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchVerticalsUseCase.kt

```diff
@@ -0,0 +1,51 @@
+package org.wordpress.android.ui.sitecreation.usecases
+
+import kotlinx.coroutines.experimental.suspendCancellableCoroutine
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.VerticalActionBuilder
+import org.wordpress.android.fluxc.store.VerticalStore
+import org.wordpress.android.fluxc.store.VerticalStore.FetchVerticalsError
+import org.wordpress.android.fluxc.store.VerticalStore.FetchVerticalsPayload
+import org.wordpress.android.fluxc.store.VerticalStore.OnVerticalsFetched
+import org.wordpress.android.fluxc.store.VerticalStore.VerticalErrorType.GENERIC_ERROR
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+
+/**
+ * Transforms EventBus event to a coroutines.
+ *
+ * The client may dispatch multiple requests, but we want to accept only the latest one and ignore all others.
+ * We can't rely just on job.cancel() as the FetchVerticalsPayload may have already been dispatched and FluxC will
+ * return a result.
+ */
+class FetchVerticalsUseCase @Inject constructor(
+    val dispatcher: Dispatcher,
+    @Suppress("unused") val verticalStore: VerticalStore
+) {
+    /**
+     * Query - Continuation pair
+     */
+    private var pair: Pair<String, Continuation<OnVerticalsFetched>>? = null
+
+    suspend fun fetchVerticals(query: String): OnVerticalsFetched {
+        return suspendCancellableCoroutine { cont ->
```

Comment:
Does this need to be cancellable if we are not cancelling it anywhere?

Reply:
It turns out we are cancelling it, just not where I was expecting we do. I am keeping the comment for context.